### ISSUE TITLE
s3api deleteObjects cannot be used to delete a specific version of an object

### ISSUE THREAD

**Issue by demurray** [localstack/localstack#2077]
The [DeleteObjects](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html) endpoint takes a list of objects and/or object versions and deletes them. With my versioning-enabled bucket, I should be able to specify `VersionId` in addition to `Key` in the elements of the `Objects` array to delete specific object versions.

In Localstack, the main object is always deleted, even when a `VersionId` was specified. This means that the object version is not deleted and that a delete marker is created, when I expected the object version to be deleted and no delete marker to be created.

We use the AWS SDK for Javascript, but I can repro from the command line as follows:

```
BUCKET=test3
KEY=myKey
touch file.txt

aws --endpoint=http://localhost:4572 s3api create-bucket --bucket "$BUCKET"
aws --endpoint=http://localhost:4572 s3api put-bucket-versioning --bucket "$BUCKET" --versioning-configuration Status=Enabled

echo "===BUCKET VERSIONING:"
aws --endpoint=http://localhost:4572 s3api get-bucket-versioning --bucket "$BUCKET"

echo "===CREATING OBJECT..."
aws --endpoint=http://localhost:4572 s3api put-object --bucket "$BUCKET" --key "${KEY}" --body file.txt

echo "===LISTING OBJECT VERSIONS:"
aws --endpoint=http://localhost:4572 s3api list-object-versions --bucket "$BUCKET"

VERSION_ID=$(aws --endpoint=http://localhost:4572 s3api list-object-versions --bucket "$BUCKET" | jq ".Versions[0].VersionId" | sed -e "s/\"//g")

echo "===DELETING OBJECTS w/ Key=${KEY}, VersionId=${VERSION_ID}..."
aws --endpoint=http://localhost:4572 s3api delete-objects --bucket "$BUCKET" --delete "Objects=[{Key=${KEY},VersionId=${VERSION_ID}}]"

echo "===LISTING OBJECT VERSIONS (should be no versions and no delete markers):"
aws --endpoint=http://localhost:4572 s3api list-object-versions --bucket "$BUCKET"
```

Running this prints out the following:

```
===BUCKET VERSIONING:
{
    "Status": "Enabled"
}
===CREATING OBJECT...
{
    "VersionId": "2e466f8b-83a9-42bd-b49c-a6671f2ca35f",
    "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\""
}
===LISTING OBJECT VERSIONS:
{
    "Versions": [
        {
            "LastModified": "2020-02-21T20:12:20.455Z",
            "VersionId": "2e466f8b-83a9-42bd-b49c-a6671f2ca35f",
            "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
            "StorageClass": "STANDARD",
            "Key": "myKey",
            "Owner": {
                "DisplayName": "webfile",
                "ID": "75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a"
            },
            "IsLatest": true,
            "Size": 0
        }
    ]
}
===DELETING OBJECTS w/ Key=myKey, VersionId=2e466f8b-83a9-42bd-b49c-a6671f2ca35f...
{
    "Deleted": [
        {
            "Key": "myKey"
        }
    ]
}
===LISTING OBJECT VERSIONS (should be no versions and no delete markers):
{
    "DeleteMarkers": [
        {
            "Owner": {
                "DisplayName": "webfile",
                "ID": "75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a"
            },
            "IsLatest": true,
            "VersionId": "beb79270-4430-4968-bc83-37a7a683b3e1",
            "Key": "myKey",
            "LastModified": "2020-02-21T20:12:22.094Z"
        }
    ],
    "Versions": [
        {
            "LastModified": "2020-02-21T20:12:20.455Z",
            "VersionId": "2e466f8b-83a9-42bd-b49c-a6671f2ca35f",
            "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
            "StorageClass": "STANDARD",
            "Key": "myKey",
            "Owner": {
                "DisplayName": "webfile",
                "ID": "75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a"
            },
            "IsLatest": false,
            "Size": 0
        }
    ]
}
```

When I run the same commands against AWS, the `Deleted` array from the output of `delete-objects` includes the `VersionId` that was passed in, and the final call to `list-object-versions` returns no versions or delete markers.

**Comment by whummer** [comment]
Thanks for the detailed report @demurray . This issue should be fixed in #2087 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks

**Comment by demurray** [comment]
Thanks @whummer. I've confirmed that `latest` is properly deleting versions.
